src/main/java/org/gridsuite/study/server/service/RemoteServicesInspector.java

```diff
@@ -106,15 +119,28 @@ public CompletableFuture<Boolean> isServerUp(RemoteServicesProperties.Service se
      * @implNote retrieve data in parallel to optimize requests time
      */
     @SuppressWarnings("unchecked") //.toArray(...) generics cause "Generic array creation) problem
-    public Map<String, JsonNode> getServicesInfo() {
-        final CompletableFuture<Entry<String, JsonNode>>[] resultsAsync = Arrays.stream(RemoteServiceName.values())
-                .parallel()
+    public Map<String, JsonNode> getServicesInfo(@Nullable FrontService viewFilter) throws PartialResultException {
+        final CompletableFuture<Entry<String, JsonNode>>[] resultsAsync = Optional.ofNullable(viewFilter)
+                .map(srv -> remoteServicesProperties.getRemoteServiceViewFilter().get(srv))
```

Comment:
the "view" is definitely not a "srv"

Reply:
that one I agree :)

Reply:
What does "srv" mean ? Server ? Service ?